src/ol/renderer/layerrenderer.js

```diff
@@ -95,6 +97,33 @@ ol.renderer.Layer.prototype.handleImageChange = function(event) {
 
 
 /**
+ * Load the image if not already loaded, and manage the
+ * registration/deregistration of the image change listener.
+ * @param {ol.ImageBase} image Image.
+ * @return {boolean} `true` if the image is already loaded, `false`
+ *     otherwise.
+ * @protected
+ */
+ol.renderer.Layer.prototype.loadImage = function(image) {
+  var imageState = image.getState();
+  if (imageState == ol.ImageState.LOADED ||
+      imageState == ol.ImageState.ERROR) {
+    goog.events.unlisten(image, goog.events.EventType.CHANGE,
+        this.handleImageChange_, false, this);
```

Comment:
It's a bit odd to see this unregistration in a load method.  This relies on `loadImage` getting called multiple times?